ui/src/menu.test.tsx

```diff
@@ -0,0 +1,36 @@
+import Button from "@mui/material/Button";
+import MenuItem from "@mui/material/MenuItem";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import React from "react";
+import { useMenu } from "./menu";
+
+function MenuButton(props: { onClick: () => void }) {
+  const [menu, show] = useMenu({
+    children: [
+      <MenuItem key="1" onClick={props.onClick}>
+        menu-item
+      </MenuItem>,
+    ],
+  });
+  return (
+    <>
+      <Button onClick={show}>show</Button>
+      {menu}
+    </>
+  );
+}
+
+test("show/hide menu", async () => {
+  const onClick = jest.fn();
+  render(<MenuButton onClick={onClick} />);
```

Comment:
naming here is a bit weird, because normally a button's onClick is when you click on the button, but in this case its on the menu item being clicked.  maybe onMenuItem?

Reply:
Done